cosmos/module_client.go

```diff
@@ -232,23 +235,39 @@ func (mc *ModuleClient) StreamExecution(ctx context.Context, req *api.StreamExec
 	loop:
 		for {
 			select {
-			case hash := <-stream:
-				exec, err := mc.GetExecution(hash)
-				if err != nil {
-					errC <- err
-					break
+			case event := <-eventStream:
+				attrHash := fmt.Sprintf("%s.%s", execution.EventType, execution.AttributeKeyHash)
+				attrs := event.Events[attrHash]
+				alreadySeeHashes := make(map[string]bool)
+				for _, attr := range attrs {
+					// skip already see hash. it deduplicate same execution in multiple event.
+					if alreadySeeHashes[attr] {
+						continue
+					}
+					alreadySeeHashes[attr] = true
```

Comment:
tiny issue but maybe this should be done only when everything is ok so after we get the execution, before the `Filter.Match`

Reply:
The filter match is based on the date of the execution, so even if it not the right event, the data of the execution will be the same.